lib/mutant/matcher/method.rb

```diff
@@ -120,13 +120,24 @@ def subject_config(node)
         def matched_view
           return if source_location.nil?
 
+          # This is a performance optimization when using --since to avoid the cost of parsing
+          # every source file that could possibly map to a subject. A more fine-grained filtering
+          # takes places later in the process.
+          return unless relevant_source_file?
+
           ast
             .on_line(source_line)
             .select { |view| view.node.type.eql?(self.class::MATCH_NODE_TYPE) && match?(view.node) }
             .last
         end
         memoize :matched_view
 
+        def relevant_source_file?
+          diffs = env.config.matcher.diffs
+
+          diffs.empty? ? true : diffs.any? { |diff| diff.touches_path?(source_path) }
```

Comment:
It occurs to me that this should maybe be `all?` based on how the subject filters work. It seems like the subject filters have an `all?` block so that you only get the intersection of changes when you supply multiple `--since` flags. I find that slightly surprising because I would expect to get the union of every `--since` flag if I supplied multiple but I also can't think of a case where I would supply multiple. The `any?` is _correct_ in the sense that the subject filter would remove everything else later that needs to be removed anyway but if you think this should be `all?` instead I'm happy to change it because in theory that would be fewer cases to parse.

Reply:
I'll go for `#all?` semantics here to have symmetry with subject filters.